molecule/interfaces_assignments/converge.yml

```
@@ -0,0 +1,54 @@
+---
+- name: converge
+  hosts: all
+  become: true
+  tasks:
+    - name: Test interface lan, device em1 description update
+      puzzle.opnsense.interfaces_assignments:
+        identifier: "lan"
+        device: "em1"
+        description: "LAN1"
+
+    - name: Test interface wan, device update from em2 to em3
+      puzzle.opnsense.interfaces_assignments:
+        identifier: "wan"
+        device: "em3"
+        description: "wan_interface"
+      register: device_already_assigned_result
+      ignore_errors: yes
+
+    - name: "Verify that the device update failed since the device is already assigned"
+      ansible.builtin.assert:
+        that:
+          - device_already_assigned_result is failed
+        fail_msg: "This device is already assigned, please assign this unassign this device first"
+        success_msg: "This device is already assigned, please assign this unassign this device first"
+
+    - name: Test unknown identifier to used device description update
+      puzzle.opnsense.interfaces_assignments:
+        identifier: "unknown_identifier"
+        device: "em1"
+        description: "unknown_identifier"
+      register: unknown_identifier_result
+      ignore_errors: yes
+
+    - name: "Verify that the device update failed since the device is already assigned"
+      ansible.builtin.assert:
+        that:
+          - unknown_identifier_result is failed
+        fail_msg: "This device is already assigned, please assign this unassign this device first"
```

The `fail_msg` and `success_msg` of some assertions in this converge test should be reworded.
```suggestion
        fail_msg: "The given device identifier is not known."
```

---

Since device the device `em1` is already assigned to the interface `lan`, this error is actually correct. Furthermore, the module supports assigning of unknown identifiers, as long as they are not assigned to a device.

---

I see, thanks for the clarification and the rewording 👍🏼